src/js/models/layerstructure.js

```diff
@@ -1201,36 +1201,48 @@ define(function (require, exports, module) {
      */
     LayerStructure.prototype.createGroup = function (documentID, groupID, groupEndID, groupName,
         isArtboard, boundsDescriptor) {
+        // Creates the group head and end, needed for any type of group 
         var groupHead = Layer.fromGroupDescriptor(documentID, groupID, groupName, false,
                 isArtboard, boundsDescriptor),
             groupEnd = Layer.fromGroupDescriptor(documentID, groupEndID, "", true),
-            layersToMove = this.selectedNormalized.flatMap(this.descendants, this).toOrderedSet(),
-            layersToMoveIndices = layersToMove.map(this.indexOf, this),
-            layersToMoveIDs = collection.pluck(layersToMove, "id"),
-            groupHeadIndex = this.layers.size - layersToMoveIndices.last(),
-            newGroupIDs = Immutable.Seq([groupEndID, layersToMoveIDs, groupID]).flatten().reverse(),
-            removedIDs = collection
-                .difference(this.index, layersToMoveIDs) // Remove layers being moved
-                .reverse(), // Reverse because we want to slice from the end
+            groupHeadIndex,
+            updatedLayers,
+            newLayerStructure,
+            newIDs;
+        
+        if (isArtboard) {
+            groupHeadIndex = this.layers.size - 1;
+        } else {
+            var layersToMove = this.selectedNormalized.flatMap(this.descendants, this).toOrderedSet(),
+                layersToMoveIndices = layersToMove.map(this.indexOf, this),
+                layersToMoveIDs = collection.pluck(layersToMove, "id");
+            
+            groupHeadIndex = this.layers.size - layersToMoveIndices.last();
+            
+            var newGroupIDs = Immutable.Seq([groupEndID, layersToMoveIDs, groupID]).flatten().reverse(),
+                removedIDs = collection
+                    .difference(this.index, layersToMoveIDs) // Remove layers being moved
+                    .reverse(); // Reverse because we want to slice from the end
             newIDs = removedIDs
                 .slice(0, groupHeadIndex) // First chunk is all layers up to top most selected one
                 .concat(newGroupIDs) // Then our new group
-                .concat(removedIDs.slice(groupHeadIndex)), // Then the rest
-            updatedLayers = this.layers.withMutations(function (layers) {
+                 .concat(removedIDs.slice(groupHeadIndex)); // Then the rest
+        }
+
+        updatedLayers = this.layers.withMutations(function (layers) {
```

Comment:
Since updatedLayers and newLayerStructure aren't defined until here, it would be nicer to wait to declare them until here as well.